File: src/main/java/nl/sijpesteijn/lasforce/guice/WebModule.java
package nl.sijpesteijn.lasforce.guice;

import com.google.inject.servlet.ServletModule;
import com.sun.jersey.guice.spi.container.servlet.GuiceContainer;
import nl.sijpesteijn.lasforce.rest.*;

import java.util.HashMap;
import java.util.Map;

/**
 * @author: Gijs Sijpesteijn
 */
public class WebModule extends ServletModule {
    private static final String JERSEY_API_JSON_POJO_MAPPING_FEATURE = "com.sun.jersey.api.json.POJOMappingFeature";

    protected void configureServlets() {

//        bind(MessageWebSocketServlet.class).in(Singleton.class);
//        serve("/ws/message").with(MessageWebSocketServlet.class);
        bind(AnimationResource.class);
        bind(SequenceResource.class);
        bind(ShowResource.class);

        bind(PlaylistResource.class);
        bind(MessageResource.class);
        Map<String, String> params = new HashMap<String, String>();
        params.put(JERSEY_API_JSON_POJO_MAPPING_FEATURE, "true");

        serve("/rest/*").with(GuiceContainer.class, params);
    }
}

File: src/main/java/nl/sijpesteijn/lasforce/guice/WebModule.kt
package nl.sijpesteijn.lasforce.guice

import com.google.inject.servlet.ServletModule
import com.sun.jersey.guice.spi.container.servlet.GuiceContainer
import nl.sijpesteijn.lasforce.endpoint.LasforceWebSocket
import nl.sijpesteijn.lasforce.endpoint.SettingsEndpoint
import nl.sijpesteijn.lasforce.laser.Laser
import nl.sijpesteijn.lasforce.laser.LaserMock
import nl.sijpesteijn.lasforce.laser.LasforceLaser
import nl.sijpesteijn.lasforce.rest.AnimationEndpoint
import nl.sijpesteijn.lasforce.services.ConnectionService
import org.codehaus.jackson.jaxrs.JacksonJsonProvider
import java.util.*
import javax.inject.Singleton

/**
 * @author Gijs Sijpesteijn
 */
const val JERSEY_API_JSON_POJO_MAPPING_FEATURE : String = "com.sun.jersey.api.json.POJOMappingFeature"

class WebModule: ServletModule() {

    override fun configureServlets() {
        bind(AnimationEndpoint::class.java)
        bind(SettingsEndpoint::class.java)

        if (System.getProperty("os.name").toLowerCase().startsWith("mac")) {
            bind(Laser::class.java).to(LaserMock::class.java)
        } else {
            bind(Laser::class.java).to(LasforceLaser::class.java)
        }

        bind(ConnectionService::class.java)
        bind(LasforceWebSocket::class.java)

        val params = HashMap<String, String>();
        params.put(JERSEY_API_JSON_POJO_MAPPING_FEATURE, "true");
        bind(JacksonJsonProvider::class.java).`in`(Singleton::class.java)
        serve("/rest/*").with(GuiceContainer::class.java, params)
    }
}